modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/recreate/steps/ProcessRecreateDestroyResult.java

```diff
@@ -30,64 +30,66 @@
     }
 
     @Override
     public void execute(DelegateExecution execution) {
         String processInstanceId = execution.getProcessInstanceId();
         Map<String, Object> variables = runtimeService.getVariables(processInstanceId);
-        String serviceId = variables.get(RecreateConstants.ID).toString();
+        UUID serviceId = (UUID) variables.get(RecreateConstants.SERVICE_ID);
+        UUID recreateOrderId = (UUID) variables.get(RecreateConstants.RECREATE_ORDER_ID);
 
-        log.info("Recreate workflow of Instance Id : {} start check destroy status",
+        log.info("Recreate workflow of instance id : {} start check destroy status.",
                 processInstanceId);
-        ServiceRecreateEntity serviceRecreateEntity =
-                recreateService.getServiceRecreateEntityById(UUID.fromString(processInstanceId));
 
         boolean isDestroySuccess =
-                deploymentEntityHandler.isServiceDestroyedSuccess(UUID.fromString(serviceId));
+                deploymentEntityHandler.isServiceDestroyedSuccess(serviceId);
 
         if (isDestroySuccess) {
-            recreateService.updateServiceRecreateStatus(serviceRecreateEntity,
-                    RecreateStatus.DESTROY_COMPLETED, OffsetDateTime.now());
             runtimeService.setVariable(processInstanceId, RecreateConstants.IS_DESTROY_SUCCESS,
                     true);
             runtimeService.setVariable(processInstanceId, RecreateConstants.DEPLOY_RETRY_NUM, 0);
-            log.info("destroy step completed for recreate order {}", processInstanceId);
+            try {
+                Thread.sleep(60 * 1000);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+            log.info("Destroy step completed for recreate order workflow with id {}",
+                    processInstanceId);
         } else {
-            recreateService.updateServiceRecreateStatus(serviceRecreateEntity,
-                    RecreateStatus.DESTROY_FAILED, OffsetDateTime.now());
-
             runtimeService.setVariable(processInstanceId, RecreateConstants.IS_DESTROY_SUCCESS,
                     false);
-            int destroyRetryNum = updateDestroyRetryNum(processInstanceId, variables);
+            int destroyRetryNum = getDestroyRetryNum(variables);
+            runtimeService.setVariable(processInstanceId, RecreateConstants.DESTROY_RETRY_NUM,
+                    destroyRetryNum + 1);
+            log.info("Process instance： {} retry to destroy task during recreation workflow. "
+                    + "RetryCount:{}", processInstanceId, destroyRetryNum);
             if (destroyRetryNum >= 1) {
                 String userId = (String) variables.get(RecreateConstants.USER_ID);
                 runtimeService.setVariable(processInstanceId, RecreateConstants.ASSIGNEE, userId);
+                serviceOrderManager.completeOrderProgress(recreateOrderId,
+                        TaskStatus.FAILED, null);
```

Comment:
Use the error message to replace null value when calling serviceOrderManager.completeOrderProgress(recreateOrderId, TaskStatus.FAILED,
null);

Reply:
fixed
like we discussed, use deploymentEntityHandler.getServiceDeploymentEntity(serviceId) to query resultMessage.